core/pkg/eval/ievaluator.go

```diff
@@ -2,25 +2,71 @@ package eval
 
 import (
 	"github.com/open-feature/flagd/core/pkg/sync"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+const (
+	FlagdProviderName        = "flagd"
+	FeatureFlagReasonKeyName = "feature_flag.reason"
+	ExceptionTypeKeyName     = "exception.type"
+	ExceptionTypeKey         = attribute.Key(ExceptionTypeKeyName)
+	FeatureFlagReasonKey     = attribute.Key(FeatureFlagReasonKeyName)
+)
+
 type AnyValue struct {
 	Value   interface{}
 	Variant string
 	Reason  string
 	FlagKey string
+	Error   error
 }
 
-func NewAnyValue(value interface{}, variant string, reason string, flagKey string) AnyValue {
+func NewAnyValue(value interface{}, variant string, reason string, flagKey string, err error) AnyValue {
 	return AnyValue{
 		Value:   value,
 		Variant: variant,
 		Reason:  reason,
 		FlagKey: flagKey,
+		Error:   err,
+	}
+}
+
+func (a AnyValue) GetImpressionAttributes() []attribute.KeyValue {
+	return []attribute.KeyValue{
+		semconv.FeatureFlagProviderName(FlagdProviderName),
+		semconv.FeatureFlagKey(a.FlagKey),
+		semconv.FeatureFlagVariant(a.Variant),
+		FeatureFlagReason(a.Reason),
 	}
 }
 
+func (a AnyValue) GetReasonAttributes() []attribute.KeyValue {
+	attrs := []attribute.KeyValue{
+		semconv.FeatureFlagProviderName(FlagdProviderName),
+		FeatureFlagReason(a.Reason),
+	}
+
+	if a.HasError() {
+		attrs = append(attrs, ExceptionType(a.Error.Error()))
+	}
```

Comment:
I don't think we need this attribute in the metric. The reason should be smth like one of `STATIC|TARGET_MATCH|ERROR`

Reply:
yes this is another dimension to give more info about the error, if we do not need it, I will remove it

Reply:
removed